### PR TITLE
Feature/add features endpoint

### PR DESCRIPTION
**Pull request by dmannarino**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Make sure you are requesting to pull a topic/feature/bugfix branch (right side). Don't request your master!
- [x] Make sure you are making a pull request against the develop branch (left side). Also you should start your branch off our develop.
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

No features endpoint

Issue Number: https://gfw.atlassian.net/browse/GTC-598

## What is the new behavior?

- Features endpoint exists

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

I found that when specifying zoom level > 9 (i.e. when one is specifying a point exactly) that the SQL query didn't return any features. I tried various different permutations for the SQL and verified that PostgreSQL returns true for this manually entered query: `SELECT ST_Intersects('POINT(0.123456789 0.987654321)'::geometry, 'POINT(0.123456789 0.987654321)'::geometry);`

Ultimately I decided to workaround it by specifying a bounding circle of 1 meter for zoom levels great than 9, rather than no bounding circle. Feel free to insult and/or vote down this kludge.

### CODE REVIEW

**Comment by thomas-maschler** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- app/models/pydantic/features.py  1
- app/routes/features/features.py  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/wri/gfw-data-api/pullRequest?prid=5810518&bid=19020281)

**Comment by thomas-maschler** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- app/models/pydantic/features.py  1
- app/routes/features/features.py  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/wri/gfw-data-api/pullRequest?prid=5810518&bid=19020281)

**Comment by thomas-maschler** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/models/pydantic/features.py  1
- app/routes/features/features.py  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/wri/gfw-data-api/pullRequest?prid=5810518&bid=19020281)

**Comment by thomas-maschler** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/models/pydantic/features.py  1
- app/routes/features/features.py  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/wri/gfw-data-api/pullRequest?prid=5810518&bid=19020281)

File: tests/routes/__init__.py [hunk 1]
```diff
@@ -35,37 +38,47 @@
...
 
 
-def create_dataset(
-    test_client, dataset_name, metadata: Dict[str, Any]
-) -> Dict[str, Any]:
-    resp = test_client.put(f"/meta/{dataset_name}", json=metadata)
+async def create_dataset(dataset_name, metadata: Dict[str, Any]) -> Dict[str, Any]:
+    async with AsyncClient(
+        app=app, base_url="http://test", trust_env=False
+    ) as test_client:
```
[the remark this answers is not in the document]
Comment (thomas-maschler) [reply tests/routes/__init__.py:44]: added such fixture while fixing tests

File: app/routes/features/features.py [hunk 2]
```diff
@@ -1,112 +1,131 @@
...
+            fields = row.metadata["fields"]
+            break
+
+    return fields
+
+
+async def _features_response(rows) -> FeaturesResponse:
+    """Serialize ORM response."""
+    data = list(rows)
+    return FeaturesResponse(data=data)
```
[the remark this answers is not in the document]
Comment (dmannarino) [reply app/routes/features/features.py:131]: It seems to. I think you only need to use from_orm when you're loading into a Pydantic model, like User.from_orm(row). Here we don't have a corresponding Pydantic model to load into.

File: tests/routes/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (dmannarino) [reply tests/routes/__init__.py:44]: Thanks!

File: app/routes/features/features.py [hunk 3]
```diff
@@ -1,112 +1,131 @@
...
+    # Azimuthal equidistant projection
+    aeqd_proj = "+proj=aeqd +lat_0={lat} +lon_0={lon} +x_0=0 +y_0=0"
+    project = partial(
+        pyproj.transform, pyproj.Proj(aeqd_proj.format(lat=lat, lon=lng)), proj_wgs84
+    )
+    buf = Point(0, 0).buffer(meter)  # distance in metres
+
+    coord_list = transform(project, buf).exterior.coords[:]
+
+    return [coord_list]
```
[the remark this answers is not in the document]
Comment (dmannarino) [reply app/routes/features/features.py]: Done!

**Comment by thomas-maschler** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/models/pydantic/features.py  1
- app/routes/features/features.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/wri/gfw-data-api/pullRequest?prid=5810518&bid=19020281)

**Pull request merged**